File: danawa.py
#!/usr/bin/python

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import sys
import re


options = webdriver.ChromeOptions()
options.add_argument('headless')
options.add_argument('--no-sandbox')
options.add_argument('--disable-dev-shm-usage')
prefs = {'profile.managed_default_content_settings.images':2}
options.add_experimental_option("prefs", prefs)
driver = webdriver.Chrome('./chromedriver', chrome_options=options)

driver.implicitly_wait(4)

wait=WebDriverWait(driver, 10)

item = sys.argv[1]
page = sys.argv[2]
it = "http://search.danawa.com/dsearch.php?tab=goods&query="+item+"&page="+page
driver.get(it)

prod_list = driver.find_element_by_id('productListArea')
prod_items = prod_list.find_elements_by_class_name('prod_main_info')
price_regex = "<strong>.*?</strong>"

for prod_item in prod_items:
    prod_img = prod_item.find_element_by_tag_name('img')
    prod_name = prod_item.find_element_by_tag_name('a')
    
    prod_img_link = prod_img.get_attribute('data-original')
    prod_path = prod_name.get_attribute('href')
    product = prod_img.get_attribute('alt').encode('utf-8').strip()
    print(prod_img_link)
    print(prod_path)
    print("<span>"+product+"</span>");
    prod_price = prod_item.find_elements_by_class_name('price_sect')
    for prices in prod_price:
        price = prices.get_attribute('innerHTML').encode('ascii', 'ignore')
	price_ = re.findall(price_regex, price)
	if not price_:
		print('<strong>none</strong>')
	if price_:
    		print(price_[0])
        

driver.quit()

File: danawa_.php
<?php

	$item = $_GET['item'];
	$page = $_GET['page'];
	$item = str_replace(" ", "+", $item);
	$handler = shell_exec("./danawa.py $item $page");

	
	// 여기서 각 정보 구분해서 출력
	// 전체 handler 에서 종류별로 구분.

	$product = 0;
	$arr_data = array();
	$arr_line = explode("\n", $handler);
	foreach($arr_line as $line)
	{
		// 이미지 링크
		if(strpos($line, "img") > 0)
		{
			$product++;
			$arr_data[$product]['img'] = $line;
			$arr_data[$product]['price'] = '';
		}
		// 이동 경로
		elseif(strpos($line, "prod") > 0)
		{
			$arr_data[$product]['path'] = $line;
		}
		// 가격
		elseif(strpos($line, "strong")>0)
		{
			$arr_data[$product]['price'] = $arr_data[$product]['price']."₩".$line."<br>";
		}
		elseif(strpos($line, "span") > 0)
		{
			$arr_data[$product]['prod'] = $line;
		}
	}
	for ($row = 1; $row<$product; $row++){
		echo '<a href="'.$arr_data[$row]['path'].'" target="_blank"><li class="prod_item"><div class="item_block"><div class="item_img"><img src="'.$arr_data[$row]['img'].'" width="100%"></img></div><div class="item_title">'.$arr_data[$row]['prod'].'</div><div class="item_price">'.$arr_data[$row]['price'].'</div></div></li></a>';
	}
?>
